python_modules/dagster/dagster/core/object_store.py

```diff
@@ -89,13 +89,39 @@ def set_value(self, obj, context, runtime_type, paths):
             return self.TYPE_REGISTRY[runtime_type.name].set_object(
                 self, obj, context, runtime_type, paths
             )
+        elif runtime_type.name is None:
+            composite_overrides = {
+                t.name in self.TYPE_REGISTRY
+                for t in runtime_type.inner_types
+                if t.name in self.TYPE_REGISTRY
+            }
+            if composite_overrides:
+                check.not_implemented(
+                    'Don\'t know how to store a composite object with inner type storage '
+                    'overrides. Found inner types with overrides: {composite_overrides}.'.format(
+                        composite_overrides=', '.join([str(x) for x in composite_overrides])
+                    )
+                )
         return self.set_object(obj, context, runtime_type, paths)
 
     def get_value(self, context, runtime_type, paths):
         if runtime_type.name is not None and runtime_type.name in self.TYPE_REGISTRY:
             return self.TYPE_REGISTRY[runtime_type.name].get_object(
                 self, context, runtime_type, paths
             )
+        elif runtime_type.name is None:
+            composite_overrides = {
```

Comment:
factor in one function so that copy is shared at a minimum

Reply:
:+1: